### PR TITLE
Granular counters

### PR DESCRIPTION
**Pull request by gustavo-salazar**
On this PR I went through all the usages of `extra_fields=counters` and started using the new feature to specify particular counters to be included to reduce response times.

Additionally, there is a new script that uses `puppeteer` to visit all the combinations of filters in the browse area, and that way a local API can generate the `request.log` with all the recent URLs and we can use that to generate the next release cache.
This becomes necessary as many URLs to be cached have changed to the new granular counters format.

Also while checking this, please have a look at this [commit in the deployment scripts repo](https://github.com/ProteinsWebTeam/InterPro7-extra/commit/a057aed4090cd390c337f9961da9c20442a91dbe) which is a modification to the script that gets the URL from the live machines, to alternatively support a path as an argument and use a file instead.

### CODE REVIEW

File: src/components/Matches/FileExporter/index.tsx [hunk 1]
```diff
@@ -58,6 +59,10 @@ const FileExporter = ({
   if (focused && +focused !== 1 && customLocationDescription.taxonomy) {
     customLocationDescription.taxonomy.accession = focused;
   }
+  const counters = getNeededCountersForSubpages(secondary, primary, true);
```
Comment (matthiasblum) [inline src/components/Matches/FileExporter/index.tsx:62]: Adding a comment for future me :sweat_smile:

```suggestion
  /*
   * Not a mistake: the filtering/secondary endpoint in the client (subpage)
   * becomes the main endpoint when exporting data.
   * e.g. InterPro entry IPR000001 > Proteins
   *   client: `/entry/interpro/IPR000001/protein/uniprot`
   *   export: `/api/protein/uniprot/entry/interpro/IPR000001/`
   */
  const counters = getNeededCountersForSubpages(secondary, primary, true);
```

**Review by matthiasblum — APPROVED** [review]
Looks good to me.

> Additionally, there is a new script that uses `puppeteer` to visit all the combinations of filters in the browse area, and that way a local API can generate the `request.log` with all the recent URLs and we can use that to generate the next release cache.
> This becomes necessary as many URLs to be cached have changed to the new granular counters format.

The new script generates URLs for all the combinations of filters in the "Browse By" pages, but it does not handle subpages (e.g. `/interpro/protein/UniProt/A0A000/entry/InterPro/`, `/interpro/protein/UniProt/A0A000/entry/cathgene3d/`, etc.), right? For subpages, we will still rely on URLs from the previous cache?

> Also while checking this, please have a look at this [commit in the deployment scripts repo](https://github.com/ProteinsWebTeam/InterPro7-extra/commit/a057aed4090cd390c337f9961da9c20442a91dbe) which is a modification to the script that gets the URL from the live machines, to alternatively support a path as an argument and use a file instead.

Added two comments: [1](https://github.com/ProteinsWebTeam/InterPro7-extra/commit/a057aed4090cd390c337f9961da9c20442a91dbe#r138588388), [2](https://github.com/ProteinsWebTeam/InterPro7-extra/commit/a057aed4090cd390c337f9961da9c20442a91dbe#r138588667)

---

Unrelated to this PR, but why does the _Browse by Taxonomy_ page has the database filter on the left, but not the _Browse By Proteome_ page?

File: scripts/browse-with-puppetteer.js [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+  };
+
+  const clickCookieBanner = async () => {
+    const targetPage = page;
+    console.log('cookie banner');
+    await delay(6000);
+    await locatorRaceClickWithRetries([
+      targetPage.locator('::-p-text(I agree, dismiss this banner)'),
+    ]);
+  };
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply scripts/browse-with-puppetteer.js:78]: mhh... that's me just being lazy... the link to close the cookie banner sits exactly under the floating tooltips.
I could detect the closing button of the tooltip, and do it for every tooltip that is visible, or wait for 6 seconds after they automatically disappear.
Lazy Gustavo just put a `delay(6000)` in there :)

[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply scripts/browse-with-puppetteer.js:78]: oh I reread your comment now, I'm guessing it clicked before the toasts went away, maybe give it more time, ahem, extend my lazyness

**Comment by gustavo-salazar** [comment]
>For subpages, we will still rely on URLs from the previous cache?

Correct, but those are only calculated in the third and fourth stages of the cache warming, which we several times hadn't had the time to run, and haven't had major issues about it. I actually wonder if there is much value on running those stages anymore, particularly given that those keys are volatile and will self-remove in 5 days.

**Comment by gustavo-salazar** [comment]
> Unrelated to this PR, but why does the Browse by Taxonomy page has the database filter on the left, but not the Browse By Proteome page?

I can't remember exactly when it got removed. But I think was in one of the times we were taking out filters to unclutter the browse section.
TBH, I feel like is missing as it is available everywhere else, and it would be really easy to put back. shall we?

**Comment by matthiasblum** [comment]
> > For subpages, we will still rely on URLs from the previous cache?
> 
> Correct, but those are only calculated in the third and fourth stages of the cache warming, which we several times hadn't had the time to run, and haven't had major issues about it. I actually wonder if there is much value on running those stages anymore, particularly given that those keys are volatile and will self-remove in 5 days.

Seems sensible. Maybe we can remove these stages from the documentation/procedures (but still keep the code if we ever need it again)

> > Unrelated to this PR, but why does the Browse by Taxonomy page has the database filter on the left, but not the Browse By Proteome page?
> 
> I can't remember exactly when it got removed. But I think was in one of the times we were taking out filters to unclutter the browse section. TBH, I feel like is missing as it is available everywhere else, and it would be really easy to put back. shall we?

Yes, please. Feel free to create a JIRA issue.

**Pull request merged**